threads-api/__test__/getTokenWithProxy.test.ts

```diff
@@ -1,10 +1,15 @@
 import { ThreadsAPI } from '../src/threads-api';
 import { TIMEOUT, rawCredentials as credentials } from './utils/constants';
 import { describeIf } from './utils/describeIf';
+import * as HttpsProxyAgent from "https-proxy-agent";
+
+// Proxy server details
+var proxy = 'http://PROXYUSERNAME:PROXYPWD@PROXYHOST:PROXYPORT';
 
 describeIf(!!credentials)('getToken', () => {
   const threadsAPI = new ThreadsAPI({
     verbose: true,
+    httpsAgent: new HttpsProxyAgent.HttpsProxyAgent(proxy),
     ...credentials,
   });
```

Comment:
This should be a working test for all environments.
If you want to test the behavior when passing `httpsAgent`, I recommend you to...
- Mock `proxy` and underlying requests using `jest.spy` or similar and then check it uses the correct agent/proxy url.
- Make `proxy` injected via `process.env` in case someone wants to run this test on the real world.